From 51e5599f8444d3380bd973d95ffb1eae90e3af50 Mon Sep 17 00:00:00 2001
From: Aedan Cullen <aedan@pig.vision>
Date: Sat, 7 Nov 2020 23:59:49 -0600
Subject: [PATCH] drm/panel-simple: add pigvision ECX337A bridge panel

Signed-off-by: Aedan Cullen <aedan@pig.vision>
---
 drivers/gpu/drm/panel/panel-simple.c | 28 ++++++++++++++++++++++++++++
 1 file changed, 28 insertions(+)

diff --git a/drivers/gpu/drm/panel/panel-simple.c b/drivers/gpu/drm/panel/panel-simple.c
index 8abb31f83..b391e9fa5 100644
--- a/drivers/gpu/drm/panel/panel-simple.c
+++ b/drivers/gpu/drm/panel/panel-simple.c
@@ -537,6 +537,31 @@ static const struct panel_desc ampire_am_480272h3tmqw_t01h = {
 	.bus_format = MEDIA_BUS_FMT_RGB888_1X24,
 };
 
+static const struct drm_display_mode pigvision_ecx337a_rgb565_bridge_mode = {
+	.clock = 81000, // LVDS serializer example: ROHM BU90T82 on pigvision PVS114T1 testbed board
+	.hdisplay = 1280,
+	.hsync_start = 1280 + 38,
+	.hsync_end = 1280 + 38 + 16,
+	.htotal = 1280 + 38 + 16 + 16,
+	.vdisplay = 960,
+	.vsync_start = 960 + 14,
+	.vsync_end = 960 + 14 + 6,
+	.vtotal = 960 + 14 + 6 + 20,
+	.vrefresh = 60,
+	.flags = DRM_MODE_FLAG_NHSYNC | DRM_MODE_FLAG_NVSYNC,
+};
+
+static const struct panel_desc pigvision_ecx337a_rgb565_bridge = {
+	.modes = &pigvision_ecx337a_rgb565_bridge_mode,
+	.num_modes = 1,
+	.bpc = 6,
+	.size = {
+		.width = 0,
+		.height = 0,
+	},
+	.bus_format = MEDIA_BUS_FMT_RGB565_1X16,
+};
+
 static const struct drm_display_mode ampire_am800480r3tmqwa1h_mode = {
 	.clock = 33333,
 	.hdisplay = 800,
@@ -3112,6 +3137,9 @@ static const struct of_device_id platform_of_match[] = {
 	{
 		.compatible = "ampire,am-480272h3tmqw-t01h",
 		.data = &ampire_am_480272h3tmqw_t01h,
+	}, {
+		.compatible = "pigvision,ecx337a-rgb565-bridge",
+		.data = &pigvision_ecx337a_rgb565_bridge,
 	}, {
 		.compatible = "ampire,am800480r3tmqwa1h",
 		.data = &ampire_am800480r3tmqwa1h,
-- 
2.17.1

